### ISSUE TITLE
Feature: warn about quoted pattern argument to == and != operators in [[ ]] conditional expressions

### ISSUE THREAD

**Issue by yump** [koalaman/shellcheck#2897]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

arg="$1"
if [[ "$arg" == "-*" ]]; then
    echo "$arg is flag"
fi
```

#### Here's what shellcheck currently says:
```

```
#### Here's what I wanted or expected to see:

```sh
if [[ "$arg" == "-*" ]]; then
                ^--^ SCXXXX (warning): quoted argument will be evaluated as a literal comparison, not a pattern match.
```

This just bit me because of habitual defensive quoting.
